lib/directory-tree.js

```diff
@@ -3,63 +3,91 @@
 const FS = require('fs');
 const PATH = require('path');
 const constants = {
-	DIRECTORY: 'directory',
-	FILE: 'file'
-}
+    DIRECTORY: 'directory',
+    FILE: 'file',
+  };
+
+function safeReadDirSync(path) {
+  let dirData = {};
+  try {
+    dirData = FS.readdirSync(path);
+  } catch (ex) {
+    if (ex.code == 'EACCES')
+    //User does not have permissions, ignore directory
+        return null;
+    else throw ex;
+  }
 
-function safeReadDirSync (path) {
-	let dirData = {};
-	try {
-		dirData = FS.readdirSync(path);
-	} catch(ex) {
-		if (ex.code == "EACCES")
-			//User does not have permissions, ignore directory
-			return null;
-		else throw ex;
-	}
-	return dirData;
+  return dirData;
 }
 
-function directoryTree (path, options, onEachFile) {
-	const name = PATH.basename(path);
-	const item = { path, name };
-	let stats;
+function directoryTree(path, options, onEachFile) {
+  const name = PATH.basename(path);
+  const item = { path, name };
+  let stats;
+
+  if (!options) {
+    options = {};
+  }
+
+  try {
+    stats = FS.statSync(path);
+  }
+  catch (e) {
+    return null;
+  }
+
+  if (options.depth == undefined || options.depth == null || options.depth >= 0) {
```

Comment:
If you do the negated version of this you can return early and have the rest of the function without an extra level of indentation.